src/cross_toml.rs

```diff
@@ -0,0 +1,194 @@
+#![doc = include_str!("../docs/cross_toml.md")]
+
+use crate::errors::*;
+use crate::Target;
+use serde::Deserialize;
+use std::collections::HashMap;
+
+/// Build environment configuration
+#[derive(Debug, Deserialize, PartialEq)]
+pub struct CrossBuildEnvConfig {
+    volumes: Option<Vec<String>>,
+    passthrough: Option<Vec<String>>,
+}
+
+/// Build configuration
+#[derive(Debug, Deserialize, PartialEq)]
+pub struct CrossBuildConfig {
+    env: Option<CrossBuildEnvConfig>,
+    xargo: Option<bool>,
+    target: Option<String>,
+}
+
+/// Target configuration
+#[derive(Debug, Deserialize, PartialEq)]
+pub struct CrossTargetConfig {
+    passthrough: Option<Vec<String>>,
+    volumes: Option<Vec<String>>,
+    xargo: Option<bool>,
+    image: Option<String>,
+    runner: Option<String>,
+}
+
+/// Wrapper struct for mappings from [`Target`] to [`CrossTargetConfig`]
+///
+/// This is used to circumvent that serde's flatten and field aliases
+/// currently don't work together: <https://github.com/serde-rs/serde/issues/1504>
+#[derive(Debug, Deserialize, PartialEq)]
+pub struct CrossTargetMapConfig {
+    #[serde(flatten)]
+    inner: HashMap<Target, CrossTargetConfig>,
+}
+
+/// Cross configuration
+#[derive(Debug, Deserialize, PartialEq)]
+pub struct CrossToml {
+    #[serde(rename = "target")]
+    targets: Option<CrossTargetMapConfig>,
+    build: Option<CrossBuildConfig>,
+}
+
+impl CrossToml {
+    /// Parses the [`CrossToml`] from a string
+    pub fn from_str(toml_str: &str) -> Result<Self> {
+        let cfg: CrossToml = toml::from_str(toml_str)?;
```

Comment:
i think it would make sense to add [`serde-path-to-error`](https://github.com/dtolnay/path-to-error), but that can be done in another pr

Reply:
Good one!